.github/workflows/CI.yml

```diff
@@ -182,3 +183,65 @@ jobs:
         with:
           key: ${{ steps.build-cache.outputs.cache-primary-key }}
           path: ~/.cache
```

Comment:
Something missing from this workflow is only triggering the `release` job when the `Project.toml` version has changed. We could put that in `release` itself but I don't want developers thinking that a release is occurring during each commit push. What I would do is either make a separate job like `release-check` or you can add this check as part of `test`. The later option just reduces noise from having extra jobs.

You would set this up with a step which checks for the `Project.toml` version change and then uses job output to see if the next job needs to run ([example](https://how.wtf/run-workflow-step-or-job-based-on-file-changes-github-actions.html#run-job-based-on-file-changes-in-github-actions)). Also, worth noting that outputs from job matrices override each other if they have the same name: https://github.com/orgs/community/discussions/26639